libs/api/domains/air-discount-scheme/src/lib/discount/discount.service.ts

```diff
@@ -39,6 +39,13 @@ export class DiscountService {
     return credit >= 1
   }
 
+  processDiscount(discount: TDiscount): TDiscount {
+    if (!this.discountIsValid(discount)) {
+      discount.discountCode = ''
```

Comment:
I'd suggest setting discountCode to null instead. Feels more explicit then a magic string ""

Reply:
I agree. Done
![image](https://user-images.githubusercontent.com/77672665/212347738-14c5a8c0-75bb-4fd6-8d12-0ae75e842ee2.png)